crates/rpc/src/discv5_rpc.rs

```diff
@@ -44,23 +49,82 @@ impl Discv5ApiServer for Discv5Api {
     }
 
     /// Write an Ethereum Node Record to the routing table.
-    async fn add_enr(&self, _enr: Enr) -> RpcResult<bool> {
-        Err(RpcServeError::MethodNotFound("add_enr".to_owned()))?
+    async fn add_enr(&self, enr: Enr) -> RpcResult<bool> {
+        match self.discv5.add_enr(enr) {
+            Ok(_) => Ok(true),
+            Err(_) => Ok(false),
+        }
     }
 
     /// Fetch the latest ENR associated with the given node ID.
-    async fn get_enr(&self, _node_id: NodeId) -> RpcResult<Enr> {
-        Err(RpcServeError::MethodNotFound("get_enr".to_owned()))?
+    async fn get_enr(&self, node_id: NodeId) -> RpcResult<Enr> {
+        self.discv5
+            .find_enr(&node_id)
+            .ok_or_else(|| RpcServeError::Message("ENR not found".to_string()).into())
     }
 
     /// Delete Node ID from the routing table.
-    async fn delete_enr(&self, _node_id: NodeId) -> RpcResult<bool> {
-        Err(RpcServeError::MethodNotFound("delete_enr".to_owned()))?
+    async fn delete_enr(&self, node_id: NodeId) -> RpcResult<bool> {
+        Ok(self.discv5.remove_node(&node_id))
     }
 
     /// Fetch the ENR representation associated with the given Node ID.
-    async fn lookup_enr(&self, _node_id: NodeId) -> RpcResult<Enr> {
-        Err(RpcServeError::MethodNotFound("lookup_enr".to_owned()))?
+    async fn lookup_enr(&self, node_id: NodeId) -> RpcResult<Enr> {
+        for enr in self.recursive_find_nodes(node_id).await? {
+            if enr.node_id() == node_id {
+                return Ok(enr);
+            } else if let Some(enr) = self
+                .find_node(enr, vec![256])
+                .await?
+                .into_iter()
+                .find(|enr| enr.node_id() == node_id)
+            {
+                return Ok(enr);
+            }
+        }
+
+        Err(RpcServeError::Message(format!("Unable to find ENR for node_id: {node_id}")).into())
+    }
+
+    /// Look up ENRs closest to the given target
+    async fn recursive_find_nodes(&self, node_id: NodeId) -> RpcResult<Vec<Enr>> {
+        match self.discv5.recursive_find_nodes(node_id).await {
+            Ok(enrs) => Ok(enrs),
+            Err(err) => Err(RpcServeError::Message(format!(
+                "Unable to send recursive_find_nodes request: {err}"
+            ))
+            .into()),
+        }
+    }
+
+    /// Send a TALKREQ request with a payload to a given peer and wait for response.
+    async fn talk_req(&self, enr: Enr, protocol: String, request: Vec<u8>) -> RpcResult<Bytes> {
+        let subnetwork = protocol
+            .parse::<Subnetwork>()
+            .map_err(|err| RpcServeError::Message(format!("Unable to parse Subnetwork: {err}")))?;
```

Comment:
```suggestion
    async fn talk_req(&self, enr: Enr, protocol: Subnetwork, request: Vec<u8>) -> RpcResult<Bytes> {
```

Couldn't we just inline the use of Subnetwork instead of parsing it here?